pkg/controller/mobileapp/create.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mobileapp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+)
+
+var (
+	ErrNoAppIDSpecified = errors.New("OS selection requires app id")
+	ErrBadSHALength     = errors.New("sha wasn't 95 characters")
+	ErrBadSHAFormat     = errors.New("sha must be of the form AA:...:BB")
+)
+
+type FormData struct {
+	Name  string          `form:"name"`
+	OS    database.OSType `form:"os"`
+	AppID string          `form:"appID"`
+	SHA   string          `form:"sha"`
+}
+
+// cleanSHA cleans an input string.
+func cleanSHA(input string) string {
+	input = strings.ToUpper(input)
```

Comment:
This is pretty expensive... it probably doesn't matter for this use case, but ReplaceAll is an O(n) op and we're calling it 5x

Reply:
Also, it feels like `clean` and `verify` should be the same function.